### PR TITLE
Start on classification: grid search, cross val and normal testing

### PR DESCRIPTION
**Pull request by kwendel**
Notes for myself:
- [x] Match feature names to information gain + sort ascending
- [x] Define metric for grid search - which metric do we refit on?
- [x] Fix classnames in classification report - is 0 clickbait?
- [x] Fix usage of all metrics for normal testing
- [x] Define more classifiers - AdaBoost, NaiveBays, XDGBoost
- [x] Feature scaling??
- [x] Class sizes are unequal

### CODE REVIEW

**Comment by kwendel** [comment]
**Metric discussion** 
The authors do not explain on what metric they optimize during grid search. For now, my selecting will be the F1 metric as it is the harmonic mean between precision and recall, and thus it combines our evaluation metrics. To be discussed with the others...

**Comment by Emiel94** [comment]
LGTM, only merge conflicts have to be resolved first

File: code/classification/Classifiers.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+metrics = ['accuracy', 'precision', 'f1', 'roc_auc', 'recall']
+classnames = ['no-clickbait', 'clickbait']
+
+
+class Classifiers:
+    def __init__(self, feature_df: pd.DataFrame, labels, classifiers):
+        self.df = feature_df
+        # Extract the dataframe as a numpy array
+        self.data = feature_df.to_numpy()
```
Comment (svenvanhal) [inline code/classification/Classifiers.py:37]: Ah I can just output this from the feature extract class 😅

File: code/classification/Classifiers.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        pvals = chi2(data, labels)
+
+        print("Chi2 stats of whole dataset")
+        print(pvals)
+
+        return pvals
+
+    def standard_scaling(self):
+        # Scale features to N(0,1) -> xi - mean(x) / std(x)
+        # THIS ASSUME THAT THE DATA IS NORMAL DISTRIBUTED!!
```
Comment (svenvanhal) [inline code/classification/Classifiers.py:83]: May need to check if this is the case

File: code/classification/Classifiers.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        return scaled_features
+
+    def robust_scaling(self):
+        # Scale just like minmax but more robust against outliers
+        scaler = RobustScaler()
+        scaled_features = scaler.fit_transform(self.data, self.labels)
+
+        return scaled_features
+
+    def _get_clf_attributes(self, val):
```
Comment (svenvanhal) [inline code/classification/Classifiers.py:103]: Convention is double `__` for private methods ;p #nitpicking

**Review by svenvanhal — APPROVED** [review]

**Comment by svenvanhal** [comment]
Verrie nice, but use the feature extraction notebook from master. I'll make a merge commit, 1 sec
